File: frontend/src/types/moment.ts
export interface MomentComment {
  id: number
  momentId: number
  rootId: number
  userId: number
  userName: string
  userAvatar?: string
  replyToId: number
  replyToName?: string
  content: string
  createTime: string
}

export interface Moment {
  id: number
  userId: number
  userName: string
  userAvatar?: string
  content?: string
  images?: string[]
  location?: string
  likeCount: number
  commentCount: number
  createTime: string
  liked: boolean
  likeUsers: string[]
  comments: MomentComment[]
}

File: backend/blog-app/src/main/java/com/natsume/blog/content/entity/Moment.java
package com.natsume.blog.content.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.extension.handlers.JacksonTypeHandler;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
@TableName(value = "t_moment", autoResultMap = true)
public class Moment {

    @TableId(type = IdType.AUTO)
    private Long id;
    private Long userId;
    private String userName;
    private String userAvatar;
    private String content;

    @TableField(typeHandler = JacksonTypeHandler.class)
    private List<String> images;

    private String location;
    private Integer likeCount;
    private Integer commentCount;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;
}
